### PR TITLE
Error handler additional context

### PR DESCRIPTION
**Pull request by jkdufair**
Closes #56. See issue for example of Sentry error that can use some extra context.

Or just click this 😂  https://sentry.io/share/issue/bd7fcaf4974b4cc9b4747162193926e1/

### CODE REVIEW

**Comment by jkdufair** [comment]
This is ready for review. Provides more context by capturing `fetchConfig` and `fetchResult` and passing them to sentry when any fetch exception occurs. [This Sentry error](https://sentry.io/organizations/purdue-university/issues/1991196095/?project=1402389&query=is%3Aunresolved) is an example that includes the additional context.

Note the additional sections on that error page below for "Fetch Config" and "Fetch Result"

While we do get some of this info in the existing context, it a) makes the backend error message/stack trace much easier to read and b) provides headers to help diagnose errors, particularly "Failed to fetch". These may be useful according to [this StackOverflow link](https://stackoverflow.com/questions/49343024/getting-typeerror-failed-to-fetch-when-the-request-hasnt-actually-failed)

File: src/fetchSaga.ts [hunk 1]
```diff
@@ -502,7 +503,7 @@ export function* fetchDataLoop(action: FetchAction): SagaIterator {
 			yield delay(action.period)
 		} while (true)
 	} catch (error) {
-		errorFunction(error.message)
+		errorHandler(error.message)
```
Comment (tragessere) [inline src/fetchSaga.ts:506]: @jkdufair Should this just be passing in the error too? I noticed the other call removed the `.message` part, and the typing says it takes an `Error` object.

Comment (jkdufair) [reply src/fetchSaga.ts]: Nice catch. Remembered and then forgot this was  there. Surprised it let me build!

Comment (tragessere) [reply src/fetchSaga.ts]: The error object is `any` typed, so typescript can't help much there.

**Review by tragessere — APPROVED** [review]

**Comment by builtbywill** [comment]
@jkdufair I think what you have is looking pretty good. I was looking at `fetchService` and am now wondering if we should use more of the `Response` object that comes back from calling `fetch()`?

navigating to the dom type file shows me this

```
interface Response extends Body {
    readonly headers: Headers;
    readonly ok: boolean;
    readonly redirected: boolean;
    readonly status: number;
    readonly statusText: string;
    readonly trailer: Promise<Headers>;
    readonly type: ResponseType;
    readonly url: string;
    clone(): Response;
}
```

**Comment by builtbywill** [comment]
@jkdufair I'm thinking at very least response headers might be more useful than request headers, since we are the ones constructing those. 

It feels like it wouldn't hurt to include those and the other response properties, except probably `trailer`

**Comment by builtbywill** [comment]
@jkdufair also it might be a good idea to define an interface/type for the `result` from `doFetch`, especially if we update that object to include more data

**Comment by jkdufair** [comment]
@builtbywill Great suggestions. I added the additional properties and declared an interface. Let me know what you think.

We do indeed construct request headers and such, but they seem to be heretofore unavailable in some Sentry errors. I don't think we'd necessarily have access to `queryParams`, `routeParams`, even `method` in some cases without passing the fetchConfig. We can infer some of the fetchConfig, but I feel like it'll be nice to have the whole thing when we are trying to reconstruct the problem?

File: src/fetchService.ts [hunk 2]
```diff
@@ -91,11 +91,17 @@ export function* doFetch(config: FetchConfig): SagaIterator {
 	}
 
 	// construct a subset of the response object to return
-	const result = {
-		ok: response.ok,
-		status: response.status,
+	const result: FetchResult = (({ headers, ok, redirected, status, statusText, type, url, bodyUsed }) => ({
```
Comment (builtbywill) [inline src/fetchService.ts:94]: @jkdufair why is this an inline function? 🤔

Comment (builtbywill) [reply src/fetchService.ts:94]: just to destructure the response? i guess that works

**Comment by builtbywill** [comment]
@jkdufair fyi please still do follow git-flow pattern (feature -> develop, etc.)

**Comment by builtbywill** [comment]
@jkdufair I am getting a test failure ATM

```
Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/fetchSaga.test.ts:490:5 - error TS2322: Type 'Error' is not assignable to type 'string'.

    490                         errorOutput = errorMessage
                                ~~~~~~~~~~~
```

File: src/fetchService.ts (hunk 2 above)
Comment (jkdufair) [reply src/fetchService.ts:94]: Seemed to be the best way to not have to duplicate the names of the props we want. 🤷

**Comment by jkdufair** [comment]
@builtbywill Test is fixed regardless

**Review by builtbywill — APPROVED** [review]
@jkdufair Looks good!

**Pull request merged**